src/viscosity.cpp

```diff
@@ -84,11 +92,23 @@ void viscosity_kernel(int x_min, int x_max, int y_min, int y_max,
 //  viscosity.
 void viscosity(global_variables &globals) {
 
+	#if FLUSH_BUFFER
+	globals.hostToDevice();
+	#endif
+
+
 	for (int tile = 0; tile < globals.config.tiles_per_chunk; ++tile) {
 		tile_type &t = globals.chunk.tiles[tile];
-		viscosity_kernel(t.info.t_xmin, t.info.t_xmax, t.info.t_ymin, t.info.t_ymax,
-		                 t.field.celldx, t.field.celldy, t.field.density0,
-		                 t.field.pressure, t.field.viscosity, t.field.xvel0,
-		                 t.field.yvel0);
+		viscosity_kernel(globals.use_target,
+		                 t.info.t_xmin,
+		                 t.info.t_xmax,
+		                 t.info.t_ymin,
+		                 t.info.t_ymax,
+		                 t.field);
 	}
+
+	#if FLUSH_BUFFER
+	globals.deviceToHost();
```

Comment:
What's this for? Can we add a comment please.

Reply:
Documented in https://github.com/UoB-HPC/cloverleaf_openmp_target/pull/2/commits/030edc6cb280ba3f773a581bbd53393e3d9828b3